src/utilities/middleware.browser.spec.ts

```diff
@@ -2,50 +2,73 @@ import { expect, test, vi } from 'vitest'
 import { RouteMiddleware } from '@/types'
 import { RouterPushError, RouterRejectionError, RouterReplaceError } from '@/types/errors'
 import { Route } from '@/types/routes'
+import { RouterRoute } from '@/utilities/createRouterRoute'
+import { createRouterRouteQuery } from '@/utilities/createRouterRouteQuery'
 import { executeMiddleware } from '@/utilities/middleware'
-import { resolveRoutes } from '@/utilities/resolveRoutes'
 import { component } from '@/utilities/testHelpers'
 
 test('calls middleware with correct routes', () => {
   const middleware = vi.fn()
+
   const routeA = {
     name: 'routeA',
     path: '/routeA',
     component,
     middleware,
   } as const satisfies Route
 
+  const routerRouteA: RouterRoute = {
+    matched: routeA,
+    matches: [routeA],
+    name: routeA.name,
+    query: createRouterRouteQuery(),
+    params: {},
+  }
+
   const routeB = {
     name: 'routeB',
     path: '/routeB',
     component,
   } as const satisfies Route
 
-  const [resolvedA, resolvedB] = resolveRoutes([routeA, routeB])
+  const routerRouteB: RouterRoute = {
+    matched: routeB,
+    matches: [routeB],
+    name: routeB.name,
+    query: createRouterRouteQuery(),
+    params: {},
+  }
 
-  executeMiddleware({ to: resolvedA, from: resolvedB })
+  executeMiddleware({ to: routerRouteA, from: routerRouteB })
 
   expect(middleware).toHaveBeenCalledOnce()
 
   const [to, { from }] = middleware.mock.lastCall
-  expect(to).toMatchObject(resolvedA)
-  expect(from).toMatchObject(resolvedB)
+  expect(to).toMatchObject(routerRouteA!)
+  expect(from).toMatchObject(routerRouteB!)
```

Comment:
pretty sure these aren't necessary

```suggestion
  expect(to).toMatchObject(routerRouteA)
  expect(from).toMatchObject(routerRouteB)
```